fe1-web/src/features/popcha/network/messages/PopchaAuthMsg.ts

```diff
@@ -0,0 +1,78 @@
+import { ActionType, MessageData, ObjectType } from 'core/network/jsonrpc/messages';
+import { validateDataObject } from 'core/network/validation';
+import { Hash, ProtocolError } from 'core/objects';
+
+/**
+ * Data sent to authenticate a user
+ */
+export class PopchaAuthMsg implements MessageData {
+  public readonly object: ObjectType = ObjectType.POPCHA;
+
+  public readonly action: ActionType = ActionType.AUTH;
+
+  public readonly client_id: string;
+
+  public readonly nonce: string;
+
+  public readonly identifier: Hash;
+
+  public readonly identifier_proof: Hash;
+
+  public readonly state: string | undefined;
+
+  public readonly response_mode: string | undefined;
+
+  public readonly popcha_address: string;
```

Comment:
It could be nice to add comments to understand what is really contained in the message. It may be obvious for you, but may not be for the people that do not know about the Popcha's design. `state` , `response_mode` or `identifier_proof` are not really self-explained in my opinion.